iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/mapreduce/IcebergInputFormat.java

```diff
@@ -549,4 +534,43 @@ private static Schema schemaWithoutConstantsAndMeta(Schema readSchema, Map<Integ
     }
   }
 
+  public static class VirtualColumnAwareIterator<T> implements CloseableIterator<T> {
```

Comment:
nit: maybe move this to the IcebergAcidUtil, so we do not have to use magic numbers, like `4`?